defnet/http_server.lua

```diff
@@ -140,7 +140,7 @@ function M.create(port)
 		local ok, err = pcall(function()
 			local request, err = receive_request(conn)
 			if err then
-				return
+				return err
```

Comment:
Shouldn't this be `return nil, err` ?

Reply:
What would `nil` represent here? Bear in mind, that `pcall` returns `true` plus everything returned by the function if no error happened or `false, err` if it caught an error during the callback. More about it [here](https://www.lua.org/pil/8.4.html). It might actually be a better practice to throw an error inside `pcall`, so it does everything for us.

Reply:
Oh, you're right. Yes, throwing a Lua error might actually be more appropriate.

Reply:
After some thought and experiments I now don't think throwing error is the correct thing to do because `err` then becomes `defnet/http_server.lua:143: closed` and we now need to strip it down to only the error message.

Reply:
Never mind, I found out that it is possible to use error(err, 0) exactly for this.